src/CMakeLists.txt

```diff
@@ -28,7 +28,11 @@ include(ROCMInstallTargets)
 include(ROCMPackageConfigHelpers)
 include(RegisterOp)
 include(CheckCXXLinkerFlag)
- 
+
+if(WIN32)
+    # Due to compilation crashing, we need to use type-erased matchers on Windows.
+    add_compile_definitions($<$<COMPILE_LANGUAGE:C,CXX>:MIGRAPHX_USE_TYPE_ERASED_MATCHERS=1>)
```

Comment:
I dont think the generator expression is needed here since it applies to both C and C++. Also this could be applied to the migraphx target since this is the component that uses the define: `target_compile_definitions(migraphx INTERFACE MIGRAPHX_USE_TYPE_ERASED_MATCHERS=1)`.

Reply:
The problem we observe - clang compiler crashing due out of memory - is when we compile CPU and GPU targets only. If I switch the option only for the mentioned targets, the compilation of `migraphx` takes a very long time, especially the linking stage. When I enabled that switch globally, the compilation of crashing targets passed, and the overall compilation time significantly shortened.

Reply:
Ah sorry, it should be `target_compile_definitions(migraphx PUBLIC MIGRAPHX_USE_TYPE_ERASED_MATCHERS=1)`, this will set it for any cmake target that uses the `migraphx/matchers.hpp` header including `migraphx` itself.